src/Compat.jl

```diff
@@ -607,6 +607,36 @@ if VERSION < v"1.5.0-DEV.438" # 0a43c0f1d21ce9c647c49111d93927369cd20f85
     Base.startswith(s) = Base.Fix2(startswith, s)
 end
 
+# https://github.com/JuliaLang/julia/pull/37517
+if VERSION < v"1.6.0" # TODO: specify the version when JuliaLang/julia#37517 is merged
+    # https://github.com/JuliaLang/julia/pull/35980
+    if VERSION < v"1.6.0-DEV.85"
+        const ComposedFunction = let h = identity ∘ convert
+            Base.typename(typeof(h)).wrapper
+        end
+        @eval ComposedFunction{F,G}(f, g) where {F,G} =
+            $(Expr(:new, :(ComposedFunction{F,G}), :f, :g))
+        ComposedFunction(f::F, g::G) where {F,G} = ComposedFunction{F,G}(f, g)
```

Comment:
Looking at the code in Base:
```suggestion
        ComposedFunction(f, g) = ComposedFunction{Core.Typeof(f),Core.Typeof(g)}(f, g)
```
I think this is better if `f` or `g` are types. For e.g. `f==Int`, `F` would be `DataType` but `Core.Typeof(f)` gives `Type{Int}`.